base/libgit2/reference.jl

```diff
@@ -169,31 +169,31 @@ function lookup_branch(repo::GitRepo,
     err = ccall((:git_branch_lookup, :libgit2), Cint,
                  (Ptr{Ptr{Void}}, Ptr{Void}, Ptr{UInt8}, Cint),
                   ref_ptr_ptr, repo.ptr, branch_name, branch_type)
-    if err == Int(Error.ENOTFOUND)
-        return nothing
-    elseif err != Int(Error.GIT_OK)
-        if ref_ptr_ptr[] != C_NULL
+    if err != Int(Error.GIT_OK)
+        if err == Int(Error.ENOTFOUND)
+            return Nullable{GitReference}()
+        elseif err != Int(Error.ENOTFOUND) && ref_ptr_ptr[] != C_NULL
             close(GitReference(repo, ref_ptr_ptr[]))
         end
         throw(Error.GitError(err))
     end
-    return GitReference(repo, ref_ptr_ptr[])
+    return Nullable{GitReference}(GitReference(repo, ref_ptr_ptr[]))
 end
 
 function upstream(ref::GitReference)
```

Comment:
Should document the return value, since Nullable is a bit unexpected. Or maybe call it `tryupstream`?